src/refactor/CSPHandler.py

```diff
@@ -0,0 +1,97 @@
+from multiprocessing.sharedctypes import Value
+import libcsp_py3 as libcsp
+from packetMaker import packetMaker
+from packetBreaker import packetBreaker
+from connectionManager import connectionManager
+import serial
+from system import SatelliteNodes
+
+def getCSPHandler(addr, interface, device, hmacKey, protocol = None):
+    if protocol is None:
+        return CSPHandler(addr, interface, device, hmacKey)
+    elif protocol == "UHF":
+        return UHF_CSPHandler(addr, interface, device, hmacKey)
+    else:
+        raise ValueError("Protocol {} does not exist for CSP handlers".format(protocol))
+
+class CSPHandler:
```

Comment:
If `CSPHandler` object is singular throughout runtime, try  making this class a singleton